.github/workflows/publish-and-deploy.yml

```
@@ -0,0 +1,68 @@
+name: publish and deploy
+
+on:
+  push:
+    branches: [master]
+
+jobs:
+  read_demos:
+    name: Read demos
+    runs-on: ubuntu-latest
+    outputs:
+      demos: ${{ steps.read-demos-folder.outputs.demos }}
+    env:
+      HUSKY: 0
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+      - name: Read demos children as JSON array
+        id: read-demos-folder
+        run: |
+          DEMOS=$(ls demos | jq -R -s -c 'split("\n")[:-1]')
+          echo ::set-output name=demos::${DEMOS}
+
+  container-builds:
+    needs:
+      - read_demos
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        demo: ${{ fromJSON(needs.read_demos.outputs.demos) }}
+    env:
+      HUSKY: 0
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+      - name: Build
+        uses: whoan/docker-build-with-cache-action@v5
+        with:
+          context: basistheory/${{ matrix.demo }}
+          image_name: ${{ matrix.demo }}
+          image_tag: latest
+          password: $${{secrets.DOCKERHUB_PASSWORD}}
+          push_git_tag: true
+          push_image_and_stages: false
```

Feel free to push in order to use caching
```suggestion
          push_image_and_stages: true
```